basic-auth/app.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+// const createError = require('http-errors');
+const express = require('express');
+const path = require('path');
+const cookieParser = require('cookie-parser');
+const logger = require('morgan');
+const mongoose = require('mongoose');
+
+const indexRouter = require('./routes/index');
+const usersRouter = require('./routes/users');
+const authRouter = require('./routes/auth');
+
+const app = express();
+
+const session = require('express-session');
+const MongoStore = require('connect-mongo')(session);
+
+// Link to database
+const dbName = 'basic-auth';
+mongoose.connect(`mongodb://localhost/${dbName}`);
+
+// view engine setup
+app.set('views', path.join(__dirname, 'views'));
+app.set('view engine', 'hbs');
+
+app.use(logger('dev'));
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+
+app.use(session({ // Always before using the routes!!!!!!!
+  store: new MongoStore({
+    mongooseConnection: mongoose.connection,
+    ttl: 24 * 60 * 60 // 1 day
+  }),
+  secret: 'some-string',
+  resave: true,
+  saveUninitialized: true,
+  cookie: {
+    maxAge: 24 * 60 * 60 * 1000
+  }
+}));
+
+app.use('/', indexRouter);
+app.use('/users', usersRouter);
+app.use('/auth', authRouter);
+
+app.use(function (req, res, next) {
+  app.locals.user = req.session.currentUser; // How to access currentUser from other routes????? --> order matters!!
```

Comment:
We haven't seen this one yet but we will. It's related to the views, not the routes